### PR TITLE
AGS support

### LINKED ISSUE #142: AGS enhacements
A [good amount](http://www.adventuregamestudio.co.uk/site/games/allgames/) of Steam and GOG games are using AGS. For instance, [this steam bundle](https://store.steampowered.com/bundle/677/The_Screen_7_Steam_Collection/) contains some good AGS games but they are only supported in Windows. It is trivial to run them using the cross-platform version of the AGS interpreter. That is why I implemented this feature using the `Scummvm.vala` source as example in my branch:

* https://github.com/neuromancer/GameHub/commit/fb8325590449f231dc3111f3826b6c550a5b970d
* https://github.com/neuromancer/GameHub/commit/d69f3a8914561679a266170c8b70d97659666cfb

I even added a little icon for that:

ags

Would you like a PR with this feature?

### PR DESCRIPTION
**Pull request by neuromancer**
This patch will add support to detect and run AGS adventures directly using GameHub, if the `ags` binary is found. It will close #142.

### CODE REVIEW

**Comment by tkashkin** [comment]
Can you provide file lists and contents of configs for a few of such games? How should these games be launched? Are there any useful launch options?

**Comment by neuromancer** [comment]
The games can be distributed only using [the generated .exe file](http://www.adventuregamestudio.co.uk/wiki/Distributing_your_game), but the games usually include the `winsetup.exe`/`ags-setup` file to configure the game (and save the configuration in `acsetup.cfg`). I can compile a list of examples from http://www.adventuregamestudio.co.uk/site/games/, if you want.
In the meantime, launch options are quite limited, in fact:

https://github.com/adventuregamestudio/ags/blob/master/Engine/main/main.cpp#L192-L209

In summary, `ags` or `ags executable.exe` works, if you run it from the game directory.

**Comment by neuromancer** [comment]
I sampled some games from the [Adventure Game Studio Depository](https://archive.org/details/adventuregamestudio) and I found they usually (but not always) contain the `audio.vox` file. So, I think the detection works well as it is using `winsetup.exe`/`ags-setup` and `acsetup.cfg`.
